extensions/control-plane/provision/provision-aws-s3/src/main/java/org/eclipse/edc/connector/provision/aws/s3/S3ProvisionPipeline.java

```diff
@@ -85,14 +86,17 @@ public CompletableFuture<S3ProvisionResponse> provision(S3BucketResourceDefiniti
         var s3AsyncClient = clientProvider.s3AsyncClient(rq);
         var iamClient = clientProvider.iamAsyncClient(rq);
         var stsClient = clientProvider.stsAsyncClient(rq);
-
-        var request = CreateBucketRequest.builder()
-                .bucket(resourceDefinition.getBucketName())
-                .createBucketConfiguration(CreateBucketConfiguration.builder().build())
-                .build();
-
-        monitor.debug("S3ProvisionPipeline: create bucket " + resourceDefinition.getBucketName());
-        return s3AsyncClient.createBucket(request)
+        var bucketName = resourceDefinition.getBucketName();
+        var headBucketRequest = HeadBucketRequest.builder().bucket(bucketName).build();
+        var createBucketRequest = CreateBucketRequest.builder().bucket(bucketName).createBucketConfiguration(CreateBucketConfiguration.builder().build()).build();
```

Comment:
this request object should be created only if needed, so in the `exceptionally` block